internal/provider/resource_marketplace_wellness_offering.go

```diff
@@ -18,6 +18,19 @@ import (
 	"github.com/lifeomic/terraform-provider-lifeomic/internal/gqlclient"
 )
 
+func retry[T any](fn func() (T, error), maxRetries int, delay time.Duration) (T, error) {
```

Comment:
I know resty has some retry functionality, but we don't expose the resty client anywhere and I don't want to enable retries for all requests, so this is my solution for now. Open to other ideas if you have them though

Reply:
If we ever find ourselves needing the above, we can use the prototype design pattern and implement a `Clone` method to create a child `client.Client` with specific retry logic/settings for specific requests.